packages/itwinui-react/src/core/Select/Select.tsx

```diff
@@ -70,6 +70,8 @@ export type SelectOption<T> = {
   size?: 'default' | 'large';
   /**
    * Value of the item.
+   *
+   * Pass `null` to reset the value.
    */
   value: T;
```

Comment:
non-blocking: Since we're also accepting `null`, are we expecting users to define `T` to also include `null`?

```diff
- <Select<number>
+ <Select<number | null>
  value={value}
  onChange={(value) => setValue(value)}
  placeholder='Select an option'
  options={[
    { value: 1, label: 'Option 1' },
    { value: 2, label: 'Option 2' },
    { value: 3, label: 'Option 3' },
  ]}
/>
```

Because with this type modification, `options.value` can also be `null`.

Or should we update the type of `value` to also include `null` so that users don't need to change their types?

If this comment involves too many changes, this can come in a later PR too.

Reply:
I've changed it ([`c8d4b16`](https://github.com/iTwin/iTwinUI/pull/1862/commits/c8d4b161f3706659c729225bdac81da06f5680d3)), but honestly it doesn't matter. The ideal use case doesn't even involve users explicitly passing generics to `Select`, it should be inferred automatically.

The one place it does matter is when defining the type for `useState`, where the user needs to include `null` anyway.

```jsx
const [value, setValue] = React.useState<number | null>(2);
```